src/main/java/com/azure/cosmos/kafka/connect/sink/BulkWriter.java

```diff
@@ -147,4 +168,33 @@ protected BulkOperationFailedException(int statusCode, int subStatusCode, String
             BridgeInternal.setSubStatusCode(this, subStatusCode);
         }
     }
+
+    private static class IdAndPartionKey {
+        String id;
+        PartitionKey partitionKey;
+
+        public IdAndPartionKey(String id, PartitionKey partitionKey) {
+            this.id = id;
+            this.partitionKey = partitionKey;
+        }
+
+
+        public String getId() {
+            return id;
+        }
+
+        public PartitionKey getPartitionKey() {
+            return partitionKey;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
```

Comment:
This implementation might not be fully correct, worth generating the implementation using Intellij IDE.

Reply:
Okay I will generate the equals and hashcode using intellij